lib/libcore/crucible/ptr.rs

```diff
@@ -0,0 +1,8 @@
+//! Crucible pointer intrinsics used within libcore.
+
+/// Pointer-to-usize comparison.  Unlike `ptr as usize == val`, this works on fat pointers and
+/// valid pointers too (pointers not created via integer-to-pointer casts).
+#[unstable(feature = "crucible_intrinsics", issue = "none")]
+pub fn compare_usize<T: ?Sized>(ptr: *const T, val: usize) -> bool {
+    unimplemented!("ptr::compare_usize")
```

Comment:
This function is implemented via custom translation inside crux-mir, so the Rust code in its body doesn't matter.  For functions with custom translation or overrides, so far I've only been providing bodies for `const fn`s (since they might theoretically get evaluated at compile time, in which case rustc needs actual code to run) and leaving the rest `unimplemented!`.